packages/server/src/db/index.js

```diff
@@ -767,6 +795,34 @@ async function sync(tableName, syncKey, updateCols, newRows) {
     }
 }
 
+/**
+ * Determines if a user is in an agency and that agency is in the same tenant
+ * @param  int        userId
+ * @param  int        tenantId
+ * @parm   Array[int] agencyIds
+ * @return boolean
+ * */
+async function inTenant(userId, tenantId, agencyIds) {
+    const q = knex(TABLES.users)
+        .select('users.id as user_id', 'agencies.id as agency_id')
+        .leftJoin('agencies', 'users.tenant_id', 'agencies.tenant_id')
+        .leftJoin('tenants', 'tenants.id', 'users.tenant_id')
+        .where('users.id', userId)
+        .andWhere('users.tenant_id', tenantId);
+
+    if (agencyIds.length > 1) {
+        q.andWhere((qb) => qb.whereIn('agencies.id', agencyIds));
+    } else if (agencyIds.length === 1) {
+        q.andWhere('agencies.id', agencyIds[0]);
+    } else {
+        return false;
+    }
```

Comment:
nits on this section:
 - no need to gate on one vs. multiple IDs; a `whereIn` with one item is just fine
 - I don't think you need the querybuilder callback, doesn't `q.whereIn(...)` work?
 - I think empty `agencyIds` ought to throw instead of just return false; it seems like that would always be a programming mistake?

Reply:
I didn't realize where and "andWhere" were aliases, so I thought I had to do an an andWhere => whereIn, thanks! An in clause can be a much slower query due to not activating the index or forcing a file sort, sometimes the query engine will optimize it away for a single record sometimes not.